frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnField.tsx

```diff
@@ -26,6 +26,10 @@ const MultiColumnField: React.FC<MultiColumnFieldProps> = ({
   toolTip,
   spans,
   complexFields,
+  isInlineControl,
+  rowClassName,
+  rowControlClassName,
+  rowControlLabel,
```

Comment:
(not so short) Summary of a conversation I had with Karthik over slack.

These changes raised a flag for me. 4 new props that became pass-thru props to the Row component had me wondering what was going on.

After digging through the use-case he was doing it for, it started to concern me around the separation of concerns. Inherently, seeing `...ClassName` props is not bad. Sometimes we do want to configure the slight styles of the internal components of these multi-faceted components. Especially so when they are pretty heavy like this one is.

However, when we get into the goals of this change, it was more-or-less unnecessary if we approached it from a custom-use-case point of view. The main issue here is the location and rendering visibility of the remove button. Where we want to maintain and control it via a more tailored experience to this particular PR's use-case. So we can render-prop this and save us from getting into the weeds and trying to make this component bend customly to our particular use-case.

Providing renderProps with default fallbacks is quite useful for existing components and main-use-case components. If you have a component, like this one, that has a primary goal doing a singular goal. If you start getting into props and flags that create a much bigger "flow diagram" to cover its entire set of use-cases, you end up in a maintenance nightmare where the cost of updating a component is several hours because of the use-cases that need to be tested and handled. Naturally tests help with this, but if you take a test failure as a means to investigate what happened, that in itself can prevent you from creating bugs but not reducing the time you spend chasing the change/test failure.

This is all to say, components should aim to do a singular thing. Sometimes that singular thing is "do a complex thing", which makes it very subjective how to address changes. Always aiming to keep your special use-cases outside of a component and just create "escape hatches" (ie. render props) usually serves to make things easier. Naturally, that has a limit too... Too many escape hatches is an indication of component breakdown being needed... but that's a different discussion :)

Reply:
Added a render prop `rowRenderer` to customize the look and feel from the consumer component.